packages/ui/react-ui-list/src/components/Tree/DropIndicator.tsx

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Instruction } from '@atlaskit/pragmatic-drag-and-drop-hitbox/tree-item';
+import React, { type HTMLAttributes, type CSSProperties } from 'react';
+
+import { mx } from '@dxos/react-ui-theme';
+
+export type DropIndicatorProps = {
+  instruction: Instruction;
+};
+
+type InstructionType = Exclude<Instruction, { type: 'instruction-blocked' }>['type'];
+type Orientation = 'sibling' | 'child';
+
+const edgeToOrientationMap: Record<InstructionType, Orientation> = {
+  'reorder-above': 'sibling',
+  'reorder-below': 'sibling',
+  'make-child': 'child',
+  reparent: 'child',
+};
+
+const orientationStyles: Record<Orientation, HTMLAttributes<HTMLElement>['className']> = {
+  sibling:
+    'h-[--line-thickness] left-[--terminal-radius] right-0 left-[--horizontal-indent] bg-[--line-color] before:left-[--negative-terminal-size]',
+  child: 'w-full top-0 bottom-0 border-[length:--line-thickness] before:invisible',
```

Comment:
```suggestion
    'bs-[--line-thickness] inline-start-[--terminal-radius] inline-end-0 inline-start-[--horizontal-indent] bg-[--line-color] before:inline-end-[--negative-terminal-size]',
  child: 'is-full block-start-0 block-end-0 border-[length:--line-thickness] before:invisible',
```

Reply:
handled most of these but some of them caused issues, left a todo

Reply:
Strange… I might like to pair on that as a sanity check… or maybe I got the shorthands wrong, the docs for those are here: https://stevecochrane.github.io/tailwindcss-logical/#inset